filecoin-proofs/src/api.rs

```diff
@@ -155,19 +156,19 @@ pub fn generate_post(
         sector_size,
         sector_count,
         challenge_seed,
-        &faults,
+        faults,
     );
 
     // Match the replicas to the challenges, as these are the only ones required.
     let challenged_replicas: Vec<_> = challenges
         .iter()
         .map(|c| {
-            if let Some(replica) = replicas.get(c.sector as usize) {
+            if let Some(replica) = replicas.get(u64::from(c.sector_id) as usize) {
                 Ok(replica)
             } else {
                 Err(format_err!(
-                    "Invalid challenge generated: {}, only {} sectors are being proven",
-                    c.sector as usize,
+                    "Invalid challenge generated: {:?}, only {} sectors are being proven",
+                    c.sector_id,
```

Comment:
please implement `Display` for `SectorId`

Reply:
Your wish is my command!